### ISSUE TITLE
Indent CallExpressions false-negative when arguments starts with array or object

### ISSUE THREAD

**Issue by maoueh** [eslint/eslint#8336]
**Tell us about your environment**

* **ESLint Version:** 3.18.0
* **Node Version:** 7.7.1
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
module.exports = {
    "extends": "standard",
    "plugins": [
        "standard",
        "promise"
    ]
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint indent: ["error", 2, { "CallExpression": {"arguments": 1} }] */
Object.assign(
        'one', 'two'
    )

/* eslint indent: ["error", 2, { "CallExpression": {"arguments": 1} }] */
Object.assign(
        {}, 'two',
    )
```

**What did you expect to happen?**

Both calls should report an error.

**What actually happened? Please include the actual, raw output from ESLint.**

Only the first call is reporting an error. The other one does not:

```
 npm-do eslint index.js

/private/tmp/test-eslint-call-expression-indent/index.js
  3:9  error  Expected indentation of 2 spaces but found 8  indent

✖ 1 problem (1 error, 0 warnings)
```

Both calls should report an error. Changing `{}` to `[]` is also exhibiting the problem. However, using `1`, `"1"`, `true`, `1.0`, `() => {}` correctly displays the error.

**Comment by not-an-aardvark** [comment]
Should be fixed by https://github.com/eslint/eslint/pull/7618

**Comment by maoueh** [comment]
@not-an-aardvark Will test and report, thanks 👍
